src/main/kotlin/app/revanced/cli/command/utility/InstallCommand.kt

```diff
@@ -44,20 +45,35 @@ internal object InstallCommand : Runnable {
                 }.install(Installer.Apk(apk, packageName))
             } catch (e: Exception) {
                 logger.severe(e.toString())
+                throw e
             }
 
             when (result) {
-                RootInstallerResult.FAILURE ->
+                RootInstallerResult.SUCCESS ->
+                    logger.info("Mounted the APK file")
+                RootInstallerResult.FAILURE -> {
                     logger.severe("Failed to mount the APK file")
-                is AdbInstallerResult.Failure ->
-                    logger.severe(result.exception.toString())
-                else ->
+                    throw Exception()
+                }
+                AdbInstallerResult.Success ->
                     logger.info("Installed the APK file")
+                is AdbInstallerResult.Failure -> {
+                    logger.severe(result.exception.toString())
+                    throw Exception()
+                }
+                else -> {
+                    logger.severe("Unknown installation result")
+                    throw Exception()
+                }
             }
         }
 
         runBlocking {
-            deviceSerials?.map { async { install(it) } }?.awaitAll() ?: install()
+            try {
```

Comment:
I generally find exceptions ugly/messy code. They introduce nesting and unnecessary blocks of code. Using return values seems simpler for "binary" cases. Exceptions have particular use case for raising messages or multiple kinds of cases not just true or false.

Reply:
If you pefer not to use exceptions for this control flow, then what do you propose?
Perhaps change InstallCommand/UninstallCommand to extend `Callable` instead of `Runnable` and return code from `install()` / `uninstall()`?

Reply:
Is this the expected way of returning an exit code for PicoCLI? If so, every subcommand should be subject to this change.

Reply:
Honestly no idea what the common pattern is.